Tests/test_file_tiff.py

```diff
@@ -450,6 +450,18 @@ def test_save_tiff_with_jpegtables(self):
         # Should not raise UnicodeDecodeError or anything else
         im.save(outfile)
 
+    def test_open_tiff_uint16_multiband(self):
+        """Test opening multiband TIFFs and reading all channels."""
+        base_value = 4660
+        for i in range(2, 6):
+            infile = "Tests/images/uint16_{}_{}.tif".format(i, base_value)
+            im = Image.open(infile)
+            im.load()
+            for j in range(i):
+                current_value = base_value + j
+                im.seek(j)
+                self.assertEqual(im.getpixel((0, 0)), current_value)
```

Comment:
Well I'm willing to work on this to try to get it operable as we work with scientific imaging at IntelinAir and support for uint16 and multiband imagery is important for us. It sounds like you're aware of many of the parts that will be needed to do this work. Can we open a tracking issue to discuss a plan for implementing these changes if you think they're worthwhile, as the conversation about the various uint16 and multiband support has gotten spread out over a multitude of issues so far.